modules/test/jenkins-results-parser/src/main/java/com/liferay/jenkins/results/parser/JIRAUtil.java

```diff
@@ -0,0 +1,249 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.jenkins.results.parser;
+
+import com.atlassian.jira.rest.client.api.IssueRestClient;
+import com.atlassian.jira.rest.client.api.JiraRestClient;
+import com.atlassian.jira.rest.client.api.JiraRestClientFactory;
+import com.atlassian.jira.rest.client.api.domain.Comment;
+import com.atlassian.jira.rest.client.api.domain.Issue;
+import com.atlassian.jira.rest.client.api.domain.Transition;
+import com.atlassian.jira.rest.client.api.domain.input.TransitionInput;
+import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory;
+
+import io.atlassian.util.concurrent.Promise;
+
+import java.io.IOException;
+
+import java.net.URI;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * @author Charlotte Wong
+ */
+public class JIRAUtil {
+
+	// this class should be statelsss
+
+	// move generate comment into job --
+
+	// get issues should be able to be called from PullRequest object
+
+	// cache issues with timestamp maybe -- low prio
+
+	// add JIRAIssue class(?) might make adding timestamp easier -- lowest prio
+
+	//  delete from cache on update -- low prio
+
+	// forced refresh of cache -- low prio
+
+	// issue number should be issueId
+
+	// allowed projects should be a property
+
+	public static Issue getIssue(String issueId) {
+		if (_issueCache == null) {
+			_issueCache = new ArrayList<>();
+		}
+
+		if (_timestampCache == null) {
+			_timestampCache = new ArrayList<>();
+		}
+
+		Date date = new Date();
+
+		long currentTime = date.getTime();
+
+		for (int i = 0; i < _issueCache.size(); i++) {
+			Issue issue = _issueCache.get(i);
+
+			String id = issue.getKey();
+
+			if (id.equals(issueId)) {
+				long timestamp = _timestampCache.get(i);
+
+				if ((currentTime - timestamp) > 300000) {
+					return issue;
+				}
```

Comment:
@CharlotteFrancis I'm making other changes but I wanted to point out this if statement. I believe that this will do the opposite of what you really want.